crates/pathfinder/src/sequencer.rs

```diff
@@ -1830,4 +1865,109 @@ mod tests {
+                && target != TargetChain::Invalid
+            {
+                match target {
+                    TargetChain::Mainnet => (None, sequencer::Client::new(Chain::Mainnet).unwrap()),
+                    TargetChain::Goerli => (None, sequencer::Client::new(Chain::Goerli).unwrap()),
+                    // Escaped above already
+                    TargetChain::Invalid => unreachable!(),
+                }
+            } else {
+                #[derive(serde::Deserialize, serde::Serialize)]
+                #[serde(deny_unknown_fields)]
+                struct Params {
+                    #[serde(rename = "blockNumber")]
+                    block_number: u64,
+                }
+
+                let filter = warp::get()
+                    .and(warp::path("feeder_gateway"))
+                    .and(warp::path("get_block"))
+                    .and(warp::query::<Params>())
+                    .map(move |params: Params| match params.block_number {
+                        0 => {
+                            const GOERLI_GENESIS: &str =
+                                include_str!("../fixtures/sequencer/0.9.0/block/genesis.json");
+
+                            let data = match target {
+                                TargetChain::Goerli => GOERLI_GENESIS.to_owned(),
+                                // This is a bit of a cheat, but we don't currently have a mainnet fixture and I'm hesitant to introduce one
+                                // since it requires re-organising all the fixtures.
+                                TargetChain::Mainnet => GOERLI_GENESIS.replace(
+                                    r#""block_hash": "0x7d328a71faf48c5c3857e99f20a77b18522480956d1cd5bff1ff2df3c8b427b"#,
+                                    r#""block_hash": "0x047C3637B57C2B079B93C61539950C17E868A28F46CDEF28F88521067F21E943"#,
+                                ),
+                                TargetChain::Invalid => GOERLI_GENESIS.replace(
+                                    r#"block_hash": "0x7d328"#,
+                                    r#"block_hash": "0x11111"#,
+                                ),
+                            };
+                            Response::new(data)
+                        }
+                        _ => Response::builder()
+                            .status(StatusCode::BAD_REQUEST)
+                            .body("Only supports genesis block request".to_owned())
+                            .unwrap(),
+                    });
+
+                let (addr, serve_fut) = warp::serve(filter).bind_ephemeral(([127, 0, 0, 1], 0));
+                let server_handle = tokio::spawn(serve_fut);
+                let client = sequencer::Client::with_url(
+                    reqwest::Url::parse(&format!("http://{}", addr)).unwrap(),
+                )
+                .unwrap();
+
+                (Some(server_handle), client)
+            }
+        }
```

Comment:
I couldn't quite shoe-horn these test requirements into the existing `setup` function, so I made a new one specifically for this chain test set. Probably worth refactoring these tests at some point. Although I suspect you'll end up with just warp paths..